common/djangoapps/third_party_auth/middleware.py

```diff
@@ -23,3 +27,12 @@ def get_redirect_uri(self, request, exception):
             redirect_uri = pipeline.AUTH_DISPATCH_URLS[auth_entry]
 
         return redirect_uri
+
+    def process_exception(self, request, exception):
+        """Handles specific exception raised by Python Social Auth eg HTTPError."""
+        if isinstance(exception, HTTPError):
+            messages.error(request, _('Unable to connect with the external provider, please try again'),
+                           extra_tags='social-auth')
+            return redirect("/login")
```

Comment:
what if there is `next` query parameter. Please check if we can use [`get_next_url_for_login_page`](https://github.com/edx/edx-platform/blob/master/common/djangoapps/student/helpers.py#L266)